app/controllers/api/application_controller.rb

```diff
@@ -28,4 +28,11 @@ def current_debate
   def require_current_debate
     render json: { error: 'Debate not found' }, status: :not_found unless current_debate
   end
+
+  def require_current_debate_not_closed
+    return unless current_debate
```

Comment:
I would remove this line. You are always putting this after `:require_current_debate` so the debate would be there. Also i think that checking if current debate exists is not in scope of this method